lms/djangoapps/courseware/module_render.py

```diff
@@ -844,36 +848,31 @@ def handle_xblock_callback_noauth(request, course_id, usage_id, handler, suffix=
         return _invoke_xblock_handler(request, course_id, usage_id, handler, suffix, course=course)
 
 
-def handle_xblock_callback(request, course_id, usage_id, handler, suffix=None):
+class XblockCallbackView(APIView):
     """
-    Generic view for extensions. This is where AJAX calls go.
-
-    Arguments:
-
-      - request -- the django request.
-      - location -- the module location. Used to look up the XModule instance
-      - course_id -- defines the course context for this request.
-
+    Class-based view for extensions. This is where AJAX calls go.
     Return 403 error if the user is not logged in. Raises Http404 if
     the location and course_id do not identify a valid module, the module is
     not accessible by the user, or the module raises NotFoundError. If the
     module raises any other error, it will escape this function.
     """
-    if not request.user.is_authenticated():
-        return HttpResponse('Unauthenticated', status=403)
+    authentication_classes = (SessionAuthentication, OAuth2Authentication,)
```

Comment:
You should also use `JwtAuthentication`. See http://edx-drf-extensions.readthedocs.io/en/latest/authentication.html#edx_rest_framework_extensions.authentication.JwtAuthentication.